cmd/admin-subnet-health.go

```diff
@@ -161,11 +177,86 @@ func mainAdminHealth(ctx *cli.Context) error {
 	}
 
 	if clusterHealthInfo.GetError() != "" {
-		console.Println(warnText("unable to obtain health information:"), clusterHealthInfo.GetError())
+		e = errors.New(clusterHealthInfo.GetError())
+		fatalIf(probe.NewError(e), "Unable to obtain health information")
+	}
+
+	filename := fmt.Sprintf("%s-health_%s.json.gz", filepath.Clean(aliasedURL), time.Now().Format("20060102150405"))
+	e = tarGZ(clusterHealthInfo, filename)
+	fatalIf(probe.NewError(e), "Unable to create health report file")
+
+	license := ctx.String("license")
+	if len(license) > 0 {
+		e = uploadHealthReport(aliasedURL, filename, license, ctx.Bool("dev"))
+		fatalIf(probe.NewError(e), "Unable to upload health report to Subnet portal")
+	}
+
+	return nil
+}
+
+func uploadHealthReport(alias string, filename string, license string, dev bool) error {
+	uploadURL := subnetUploadURL(alias, filename, license, dev)
+	req, e := subnetUploadReq(uploadURL, filename)
+	if e != nil {
+		return e
+	}
+
+	resp, herr := httpClient(10 * time.Second).Do(req)
+	if herr != nil {
+		return herr
+	}
+	defer resp.Body.Close()
+
+	var respBody []byte
+	respBody, e = ioutil.ReadAll(resp.Body)
+	if e != nil {
+		return e
+	}
+
+	if resp.StatusCode == http.StatusOK {
+		msg := "MinIO Health data was successfully uploaded to Subnet."
+		clusterURL, _ := url.PathUnescape(gjson.Get(string(respBody), "cluster_url").String())
+		if len(clusterURL) > 0 {
+			msg += fmt.Sprintf(" Can be viewed at: %s", clusterURL)
+		}
+		console.Infoln(msg)
```

Comment:
if `json` flag is passed, a different output should be printed. It has to be consistent with how other commands are doing it now.

```
$ mc admin user add myminio test1 test1234
Added user `test1` successfully.
$ mc admin user add myminio test1 test1234 --json
{
 "status": "success",
 "accessKey": "test1",
 "secretKey": "test1234",
 "userStatus": "enabled"
}
```

Reply:
@kannappanr the `mc admin subnet health` command currently prints the health report json in case of success (when `--json` is passed). I'm not sure if it's a good idea to change that..

Reply:
Also, the new `--license` option is not to be used with `--json` as it relies on the generated `tar.gz` report file.
I'll add validation to ensure that both these options are not passed together.